kategory/src/main/kotlin/kategory/data/StateT.kt

```diff
@@ -22,6 +22,9 @@ class StateT<F, S, A>(
         inline operator fun <reified F, S, A> invoke(noinline run: StateTFun<F, S, A>, MF: Monad<F> = monad<F>()): StateT<F, S, A> =
                 StateT(MF, MF.pure(run))
 
+        fun <F, S, A> invokeF(runF: StateTFunKind<F, S, A>, MF: Monad<F>): StateT<F, S, A> =
```

Comment:
Same story. You can't use a reified type F even if you move it to extension function (tried it). In fact, that would be changing the main intention of the function, since it would require a valid `StateT` runtime instance to be executed, so it would not be a static utility method to create a `StateT` from nothing anymore.